### PR TITLE
Solving dtype issue

### PR DESCRIPTION
**Pull request by thomas-mullor**
## Description

Add better dtype management in the inference wrapper
"to" method of the wrapper now follows torch-like signature and more general behaviour :
`my_inference_wrapper.to(device=some_device, dtype=some_dtype)`

If embeddings are passed directly to the inference wrapper, they will be casted to the adapted type (allows dealing with quantized models)

Also add light dtype management for aggregators

## Related Issue

Solves #101

## Type of Change

- 🔧 Bug fix (non-breaking change which fixes an issue)
- 🥂 Improvement (non-breaking change which improves an existing feature)

## Checklist

- [x] I've read the [`CODE_OF_CONDUCT.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/CODE_OF_CONDUCT.md) document.
- [x] I've read the [`CONTRIBUTING.md`](https://github.com/FOR-sight-ai/interpreto/blob/main/docs/contributing.md) guide.
- [x] I've successfully run the style checks using `make lint`.
- [x] I've written tests for all new methods and classes that I created and successfully ran `make test`.
- [x] I've written the docstring in Google format for all the methods and classes that I used.

### CODE REVIEW

File: interpreto/model_wrapping/inference_wrapper.py [hunk 1]
```diff
@@ -210,6 +211,14 @@ def cuda(self):
         """
         self.device = torch.device("cuda")
 
+    @property
+    def dtype(self):
+        return self.model.dtype
+
+    @dtype.setter
+    def dtype(self, dtype: torch.dtype):
+        self.model.to(dtype=dtype)
```
Comment (AntoninPoche) [inline interpreto/model_wrapping/inference_wrapper.py:228]: Is it normal to be able to set the dtype of the model?

File: tests/model_wrapping/test_inference_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    # Prepare dummy input
+    input_ids = torch.tensor([[1, 2, 3, 4]])
+
+    # Perform inference
+    outputs = wrapped_model.get_logits({"input_ids": input_ids, "attention_mask": torch.ones_like(input_ids)})
+
+    # Assertions
+    assert outputs is not None
+    assert isinstance(outputs, torch.Tensor)
+    assert outputs.shape[0] == input_ids.shape[0]  # Ensure batch size matches
```
Comment (AntoninPoche) [inline tests/model_wrapping/test_inference_wrapper.py]: I would add a test where you do a forward from `input_embeds` in `torch.float32`.

In our code, this might happen with gradient-based methods.

Because the transformation from `input_ids` to `input_embeds` is managed by the model otherwise.

File: tests/model_wrapping/test_inference_wrapper.py
Comment (AntoninPoche) [inline tests/model_wrapping/test_inference_wrapper.py:1]: Should you not also include a test with each attribution method? To ensure that the whole perturbation-inference-aggregation pipeline is robust to quantized models?

Also, at least a classification and a generation model.

A WORD granularity to be sure granularity manipulations also work with this. (Not the TOKEN default).

File: interpreto/model_wrapping/inference_wrapper.py (hunk 1 above)
Comment (thomas-mullor) [reply interpreto/model_wrapping/inference_wrapper.py:228]: It is ok in Pytorch : https://docs.pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.to
